utils/tokenbucket_test.go

```diff
@@ -0,0 +1,102 @@
+// Copyright (c) 2017-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+package utils
+
+import (
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+)
+
+type TestCase struct {
+	rate         time.Duration
+	burst        uint64
+	calls        int
+	minTotalTime time.Duration
+}
+
+// TODO(gsagula): Add a verification step that compares the waiting time with an acceptable
+// marginal error. It should just log a warning but not fail.
```

Comment:
Great test case :+1:

Blocking on approval unless you feel it's not required or want to follow up with another PR.